lib/Basics/files.cpp

```diff
@@ -88,6 +88,16 @@ struct LockfileRemover {
 
 /// @brief this instance will remove all lockfiles in its dtor
 static LockfileRemover remover;
+
+#ifdef _WIN32
+std::wstring toWString(std::string const& validUTF8String) {
+  icu::UnicodeString utf16(validUTF8String.c_str());
+  using bufferType = std::remove_pointer_t<decltype(utf16.getTerminatedBuffer())>;
+  static_assert(sizeof(std::wchar_t) == sizeof(bufferType));
+  return std::wstring(reinterpret_cast<std::wchar_t const*>(utf16.getTerminatedBuffer()));
```

Comment:
I am not familiar with ICU, but based on the `static_assert` I would assume that `getTerminatedBuffer` returns a `const wchar_t*`. So why do you need a `reinterpret_cast` here?